dso-l2/src/main/java/com/tc/objectserver/entity/ManagedEntityImpl.java

```diff
@@ -231,6 +233,21 @@ private void processSyncMessage(ServerEntityRequest sync, byte[] payload, int co
     }
   }
 
+  @Override
+  public void dumpStateTo(StateDumper stateDumper) {
+    if(activeServerEntity != null) {
+      if(activeServerEntity instanceof StateDumpable) {
```

Comment:
Probably want to add some doc here to explain that this instanceof is not a hack but actually by design, since we are currently allowing that to be an optional "opt-in" kind of capability (if we make that mandatory, we will remove this check).